yt_dlp/extractor/getcourseru.py

```diff
@@ -52,7 +52,7 @@ class GetCourseRuIE(InfoExtractor):
     _BASE_URL_RE = rf'https?://(?:(?!player02\.)[^.]+\.getcourse\.(?:ru|io)|{"|".join(map(re.escape, _DOMAINS))})'
     _VALID_URL = [
         rf'{_BASE_URL_RE}/(?!pl/|teach/)(?P<id>[^?#]+)',
-        rf'{_BASE_URL_RE}/(:?pl/)?teach/control/lesson/view\?(?:[^#]+&)?id=(?P<id>\d+)',
+        rf'{_BASE_URL_RE}/(?:pl/)?teach/control/lesson/view\?(?:[^#]+&)?id=(?P<id>\d+)',
```

Comment:
It's rather lucky, or not, that the `(?:` <-> `(:?` transposition doesn't actually constrain the match beyond what was intended: a stray `:` will also be matched, but otherwise it just introduces an unintended numbered group.

This is an extra check to be run in one's editor, since it would be difficult to tell what was meant automatically.